app/controllers/concerns/flashable.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+module Flashable
+  extend ActiveSupport::Concern
+
+  def message_for_notice(invalid_records)
+    msg = "Uploading #{upload_params[:csv_type]} succeeded"
+    msg + (invalid_records.present? ? ' with warnings:' : '.')
+  end
```

Comment:
NITPICK: So there are, as I recall a couple different versions of flash messages. One is flash[:notice] and the other is flash[:warn], and they can be setup to have different styles around them depending on how bootstrap css classes are defined. Might this be a better alternative?